matter_server/common/models.py

```diff
@@ -209,3 +211,21 @@ class CommissioningParameters:
     setup_pin_code: int
     setup_manual_code: str
     setup_qr_code: str
+
+
+@dataclass
+class MatterSoftwareVersion:
+    """Representation of a Matter software version. Return by the check_node_update command.
+
+    This holds Matter software version information similar to what is available from the CSA DCL.
+    https://on.dcl.csa-iot.org/#/Query/ModelVersion.
+    """
+
```

Comment:
maybe we can add a source here to distinguish main-net and test-net updates (or other sources maybe)

Reply:
Yeah good idea, definitely possible. I guess the question will be how do we expose that to the user (as in how can we display it in the update entity). I'd rather prefer to only add if we have actual use for it.

Reply:
Well, usecase would be to inform the use in the UI that its a beta update vs stable

Reply:
Yes, but *how* would we display that? The Update entity has no such feature currently, from what I know.

I'd imagine update source would be a enum or something, so the question also becomes how we represent extra/additional update sources.

Reply:
We don't: 
Its just to steer some custom logic in the integration to show the correct (warning) message for now).
I think what I'm trying to say is that I don't think its of interest to show the exact source to the user in HA (maybe it is in the debug UI) but we can use it to steer other logic.

So the way I see it we have 3 update providers in the code: dcl-main, dcl-test and builtin
Would be good to have that as source attribute on the model (could be an enum as well)

Reply:
Ok, I've added update source as an enum.